agent/version/gen/version-gen.go

```diff
@@ -69,20 +68,37 @@ func gitDirty() bool {
 }
 
 func gitHash() string {
-	cmd := exec.Command("git", "rev-parse", "--short=8", "HEAD")
-	hash, err := cmd.Output()
-	if err != nil {
+	readFileHash, readFileErr := os.ReadFile(filepath.Join("..", "..", "RELEASE_COMMIT"))
+	if readFileErr != nil {
+		log.Printf("Unable to read RELEASE_COMMIT file, readFileErr: %v", readFileErr)
+		cmd := exec.Command("git", "rev-parse", "--short=8", "HEAD")
+		revParseHash, revParseErr := cmd.Output()
+		if revParseErr != nil {
+			log.Printf("Failed to execute git rev-parse --short=8 HEAD, revParseErr: %v", revParseErr)
+			return "UNKNOWN"
+		}
+		hashStr := strings.TrimSpace(string(revParseHash))
+		log.Printf("Successfully retrieved short hash value from git rev-parse --short=8 HEAD, hashStr: %s",
+			revParseHash)
+		return hashStr
```

Comment:
this can be abstracted out into a separate method.

Reply:
Agreed, thanks for pointing out!